structuredheadings/plugin.js

```diff
@@ -464,6 +466,113 @@
             setStyle(editor, node, editor.config.autonumberCurrentStyle);
           }
         }
+      },
+
+      /*
+       * convertList
+       */
+      convertList: {
+        exec: function (editor) {
+
+          var selectArea = function (startElement, endElement, encompass) {
+            encompass = encompass ? encompass : false;
+            var range = editor.createRange();
+            if (encompass) {
+              range.setStartBefore(startElement, 0);
+              range.setEndAfter(endElement, 0);
+            } else {
+              range.setStart(startElement, 0);
+              range.setEnd(endElement, 0);
```

Comment:
using setStart and setEnd at the same point is placing the cursor, rather than a full selection. using setStart and setEnd on the same element, same offset is the equivalent of `'<h1>^Heading</h1>`, just enough to place cursor location to allow me to operate on that block element. Setting encompass to true would then use the setStartBefore and setEndAfter which would create the actual selection.